src/snakemake/resources.py

```
@@ -521,84 +444,493 @@
+        evaluable resources.
+        """
+        if isinstance(self._value, TBDString):
+            return self
+
+        other_val = other.value if isinstance(other, Resource) else other
+
+        if self._value is None or other_val is None:
+            return self
+
+        self_val = self.value
+        if type(self_val) != type(other_val):
```

Ruff (0.8.2)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)